### PR TITLE
Dev into Master

### PR DESCRIPTION
**Pull request by slarson**
Adds:

* Major speed improvements to loading neurons by @mwatts15 
* CodeSpeed support by @mwatts15 
* Python 3 support by @mwatts15 
* Data on neuropeptides from Bentley et al. 2016, courtesy of @clbarnes 
* Improvements to Evidence system by @mwatts15 
* Fixes to build process by @slarson 
* Many bug fixes and improvements by @mwatts15 

Deprecates

* FuXI reasoner support (library is not being supported anymore)

### CODE REVIEW

**Comment by mwatts15** [comment]
Doing a quality check before the merge.

File: PyOpenWorm/channel.py [hunk 1]
```diff
@@ -92,16 +95,16 @@ def __init__(self, name=False, **kwargs):
...
-        #Channel.DatatypeProperty('description_evidences', self)
-        #TODO: assert this in the adapter instead
-        #Channel.DatatypeProperty('expression_evidences', self)
+        Channel.DatatypeProperty('name', self)
+        Channel.DatatypeProperty('description', self)
+        Channel.DatatypeProperty('gene_name', self)
+        Channel.DatatypeProperty('gene_WB_ID', self)
+        Channel.DatatypeProperty('expression_pattern', self)
+        Channel.DatatypeProperty('proteins', self, multiple=True)
+        # TODO: assert this in the adapter instead
```
Comment (mwatts15) [inline PyOpenWorm/channel.py]: Was this a ChannelWorm-related TODO?
Added at 38199eca317505473c4f3940d39a917896760585, 27 Jul 2015

File: PyOpenWorm/evidence.py [hunk 2]
```diff
@@ -24,23 +28,33 @@ def _doi_uri_to_doi(uri):
...
     headers = {'Content-Type': 'application/json'}
     try:
-        return json.load(_url_request(url, headers))
+        data = _url_request(url, headers).read().decode('UTF-8')
+        if hasattr(data, 'charset'):
+            return json.loads(data, encoding=data.charset)
+        else:
+            return json.loads(data)
     except BaseException:
+        traceback.print_exc()
```
Comment (mwatts15) [inline PyOpenWorm/evidence.py]: Should be logging this guy rather than printing to stderr.

File: PyOpenWorm/neuron.py [hunk 3]
```diff
@@ -112,29 +114,28 @@ def _type_networkX(self):
...
 
 
 class Neighbor(P.Property):
-    multiple=True
-    def __init__(self,**kwargs):
-        P.Property.__init__(self,'neighbor',**kwargs)
+    multiple = True
+
+    def __init__(self, **kwargs):
+        P.Property.__init__(self, 'neighbor', **kwargs)
```
Comment (mwatts15) [inline PyOpenWorm/neuron.py]: should use super()

File: PyOpenWorm/neuron.py [hunk 4]
```diff
@@ -266,11 +278,11 @@ def set(self, conn, **kwargs):
            -------
            A PyOpenWorm.neuron.Connection
         """
-        #XXX: Should this create a Connection here instead?
+        # XXX: Should this create a Connection here instead?
```
Comment (mwatts15) [inline PyOpenWorm/neuron.py]: No, it shouldn't.

File: PyOpenWorm/simpleProperty.py [hunk 5]
```diff
@@ -223,34 +290,76 @@ def has_defined_value(self):
...
         cls.conf['rdf.namespace_manager'].bind(cls.__name__, cls.rdf_namespace)
+
+
+def _get_or_set(self, *args, **kwargs):
+    """ If arguments are given ``set`` method is called. Otherwise, the ``get``
+    method is called. If the ``multiple`` member is set to ``True``, then a
+    Python set containing the associated values is returned. Otherwise, a
+    single bare value is returned.
+    """
+    # XXX: checking this in advance because I'm paranoid, I guess
```
Comment (mwatts15) [inline PyOpenWorm/simpleProperty.py]: Make _get_or_set a mixin so the asserts can go into an __init__ rather than checking on every call.

**Comment by mwatts15** [comment]
+1
I'll do a release on pypi in the morning.

File: PyOpenWorm/channel.py (hunk 1 above)
Comment (slarson) [inline PyOpenWorm/channel.py]: looks like it.  why do you ask?

File: PyOpenWorm/evidence.py (hunk 2 above)
Comment (slarson) [inline PyOpenWorm/evidence.py]: is this a change we want to put in now?

File: PyOpenWorm/neuron.py (hunk 3 above)
Comment (slarson) [inline PyOpenWorm/neuron.py]: is this a change to add now?  Or make a TODO in the code for later?

File: PyOpenWorm/neuron.py (hunk 4 above)
Comment (slarson) [inline PyOpenWorm/neuron.py]: Again -- change now or add to the comment in the code?

**Pull request merged**